### PR TITLE
Fix redirection after partly failing upload

### PR DESCRIPTION
**Pull request by batikanu**
issue #385 is solved. If any resources in the file is deployed successfully than the page is redirected after upload.

### CODE REVIEW

File: src/app/frontend/deploy/deployfromfile_controller.js [hunk 1]
```diff
@@ -84,9 +84,12 @@ export default class DeployFromFileController {
     let resource = this.resource_('api/v1/appdeploymentfromfile');
     resource.save(
         deploymentSpec,
-        (savedConfig) => {
-          defer.resolve(savedConfig);  // Progress ends
-          this.log_.info('Successfully deployed application: ', savedConfig);
+        (response) => {
+          defer.resolve(response);  // Progress ends
+          this.log_.info('Deployment is completed: ', response);
+          if (response.error.length > 0) {
```
Comment (bryk) [inline src/app/frontend/deploy/deployfromfile_controller.js]: Can you test this?

File: src/app/backend/deploy.go [hunk 2]
```diff
@@ -295,12 +307,16 @@ func DeployAppFromFile(spec *AppDeploymentFromFileSpec, createObjectFromInfoFn c
...
 		Do()
 
-	return r.Visit(func(info *kubectlResource.Info, err error) error {
-		err = createObjectFromInfoFn(info)
-		if err != nil {
-			return err
+	deployedResourcesCount:= 0
+
+	err = r.Visit(func(info *kubectlResource.Info, err error) error {
+		isDeployed, err := createObjectFromInfoFn(info)
```
Comment (bryk) [inline src/app/backend/deploy.go]: Can you test this?

**Comment by bryk** [comment]
Looks very nice. We just need tests.

**Comment by batikanu** [comment]
I have added the tests PTAL

**Comment by bryk** [comment]
LGTM :)

**Pull request merged**